types/firestore.d.ts

```diff
@@ -1924,26 +1924,26 @@ declare namespace FirebaseFirestore {
     readonly bundleId: string;
 
     /**
-     * Adds a Firestore document snapshot or query snapshot to the bundle.
-     * Both the documents data and the query read time will be included in the bundle.
+     * Adds a Firestore document snapshot to the bundle. Both the documents data and the document
+     * read time will be included in the bundle.
      *
-     * @param documentSnapshot A document snapshot to add.
+     * @param documentSnapshot A `DocumentSnapshot` to add.
      * @returns This instance.
      */
-    add(documentSnapshot: DocumentSnapshot): BundleBuilder;
+    add<T>(documentSnapshot: DocumentSnapshot<T>): BundleBuilder;
 
     /**
-     * Adds a Firestore document snapshot or query snapshot to the bundle.
-     * Both the documents data and the query read time will be included in the bundle.
+     * Adds a Firestore query snapshot to the bundle. Both the query data and the query read time
```

Comment:
We are also adding the resulting documents, no?

s/query snapshot/`QuerySnapshot`

Reply:
Done.